tools/BackPopulateDeepQC.pl

```diff
@@ -0,0 +1,128 @@
+It can take in tarchiveID as an argument if only a specific .tar is to be
+processed; otherwise, all .tar in the tarchive tables are processed.
+HELP
+
+my $Usage = <<USAGE;
+
+Usage: $0 -help to list options
+
+USAGE
+
+&Getopt::Tabular::SetHelp($Help, $Usage);
+&Getopt::Tabular::GetOptions(\@opt_table, \@ARGV) || exit 1;
+
+################################################################
+################### input option error checking ################
+################################################################
+{ package Settings; do "$ENV{LORIS_CONFIG}/.loris_mri/$profile" }
+if ($profile && !@Settings::db) {
+    print "\n\tERROR: You don't have a configuration file named ".
+          "'$profile' in:  $ENV{LORIS_CONFIG}/.loris_mri/ \n\n";
+    exit 2;
+}
+
+################################################################
+######### Establish database connection ########################
+################################################################
+my $dbh = &NeuroDB::DBI::connect_to_db(@Settings::db);
+print "\nSuccessfully connected to database \n";
+
+################################################################
+######### Initialize variables #################################
+################################################################
+my $data_dir = &NeuroDB::DBI::getConfigSetting(
+                    \$dbh,'dataDirBasepath'
+                    );
+my $tarchiveLibraryDir = &NeuroDB::DBI::getConfigSetting(
+                       \$dbh,'tarchiveLibraryDir'
+                       );
+$tarchiveLibraryDir    =~ s/\/$//g;
+my ($sec,$min,$hour,$mday,$mon,$year,$wday,$yday,$isdst)
+    =localtime(time);
+my $template = "TarLoad-$hour-$min-XXXXXX"; # for tempdir
+my $TmpDir = tempdir(
+                 $template, TMPDIR => 1, CLEANUP => 1
+             );
+my @temp     = split(/\//, $TmpDir);
+my $templog  = $temp[$#temp];
+my $LogDir   = "$data_dir/logs";
+if (!-d $LogDir) {
+    mkdir($LogDir, 0770);
+}
+my $logfile  = "$LogDir/$templog.log";
+
+################################################################
+################## Instantiate MRIProcessingUtility ############
+################################################################
+my $utility = NeuroDB::MRIProcessingUtility->new(
+                  \$dbh,$debug,$TmpDir,$logfile,
+                  $LogDir,$verbose
+              );
+
+################################################################
+# Grep tarchive list for all those entries with         ########
+# NULL in ArchiveLocationPerModality                    ########
+################################################################
+
+# Query to grep all tarchive entries
+if (!defined($TarchiveID)) {
+    $query = "SELECT TarchiveID " .
+        "FROM tarchive";
+}
+# Selecting tarchiveID is redundant here but it makes the while() loop
+# applicable to both cases; when a TarchiveID is specified or not
+else {
+    $query = "SELECT TarchiveID " .
```

Comment:
in the two queries, shouldn't you check that no deep QC were already done for the tarchives?

Reply:
@cmadjar - This check is done within the `computeDeepQC` in `MRIProcessingUtility.pm` function. If there is already an entry in the `parameter_file` table and the new DeepQC result is different, that entry gets updated. @crocodoyle  and I discussed a new database structure for the Flask app a few days ago, so I'll make other relevant changes this week as well.